sdks/python/apache_beam/ml/inference/gemini_inference.py

```diff
@@ -0,0 +1,168 @@
+        this parameter sends requests for this job to the Gemini Developer API.
+        If this paramter is provided, do not set the project or location
+        parameters.
+      project: the GCP project to use for Vertex AI requests. Setting this
+        parameter routes requests to Vertex AI. If this paramter is provided,
+        location must also be provided and api_key should not be set.
+      location: the GCP project to use for Vertex AI requests. Setting this 
+        parameter routes requests to Vertex AI. If this paramter is provided,
+        project must also be provided and api_key should not be set.
+      min_batch_size: optional. the minimum batch size to use when batching
+        inputs.
+      max_batch_size: optional. the maximum batch size to use when batching
+        inputs.
+      max_batch_duration_secs: optional. the maximum amount of time to buffer 
+        a batch before emitting; used in streaming contexts.
+    """
+    self._batching_kwargs = {}
+    self._env_vars = kwargs.get('env_vars', {})
+    if min_batch_size is not None:
+      self._batching_kwargs["min_batch_size"] = min_batch_size
+    if max_batch_size is not None:
+      self._batching_kwargs["max_batch_size"] = max_batch_size
+    if max_batch_duration_secs is not None:
+      self._batching_kwargs["max_batch_duration_secs"] = max_batch_duration_secs
+
+    self.model_name = model_name
+    self.request_fn = request_fn
+
+    if api_key:
+      if project or location:
+        raise ValueError("project and location must be None if api_key is set")
+      self.api_key = api_key
+      self.use_vertex = False
+    else:
+      if project is None or location is None:
+        raise ValueError(
+            "project and location must both be provided if api_key is None")
+      self.project = project
+      self.location = location
+      self.use_vertex = True
+
+    super().__init__(
+        namespace='GeminiModelHandler',
+        retry_filter=_retry_on_appropriate_service_error,
+        **kwargs)
+
+  def create_client(self) -> genai.Client:
+    """Creates the GenAI client used to send requests. Creates a version for
+    the Vertex AI API or the Gemini Developer API based on the arguments
+    provided when the GeminiModelHandler class is instantiated.
+    """
+    if self.use_vertex:
+      return genai.Client(
+          vertexai=True, project=self.project, location=self.location)
+    return genai.Client(api_key=self.api_key)
+
+  def request(
+      self,
+      batch: Sequence[Any],
+      model: genai.Client,
+      inference_args: Optional[dict[str, Any]] = None
+  ) -> Iterable[PredictionResult]:
+    """ Sends a prediction request to a Gemini service containing a batch
+    of inputs and matches that input with the prediction response from
+    the endpoint as an iterable of PredictionResults.
+
+    Args:
+      batch: a sequence of any values to be passed to the Gemini service.
+        Should be a list of strings.
```

Comment:
Sequence[Any]: I assume we plan to support other types soon. So we use Any here, right?

Reply:
Ah yeah that was a missed line, I originally wrote this as string-only but decided it was better in the long-term to go ahead and make it more general. Let me update that docstring